gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -291,9 +366,49 @@ public int read(ByteBuffer byteBuffer) throws IOException {
         bufferedContentReadOffset = bytesToWrite;
       }
     }
+
+    if (footerContent != null && position >= footerStartOffsetInBytes && byteBuffer
+        .hasRemaining()) {
+      int bytesToWrite = min(byteBuffer.remaining(), footerContent.size());
+      int bytesToSkipInFooter = (int) (position - footerStartOffsetInBytes);
+      put(footerContent, bytesToSkipInFooter, bytesToWrite, byteBuffer);
+      position += bytesToWrite;
+      bytesRead += bytesToWrite;
+    }
+
     return bytesRead;
   }
 
+  private Integer getBytesToRead(ByteBuffer byteBuffer) {
+    Integer bytesToRead =
+        readStrategy == Fadvise.RANDOM
+            ? max(byteBuffer.remaining(), readOptions.getMinRangeRequestSize())
+            : null;
+    if (footerContent == null) {
+      return bytesToRead;
+    }
+
+    long bytesToFooterOffset = footerStartOffsetInBytes - position;
+    if (bytesToRead == null) {
```

Comment:
Isn't this `if` + `return` a mere:

```java
return bytesToRead == null ? Math.toIntExact(bytesToFooterOffset) : Math.toIntExact(min(bytesToRead, bytesToFooterOffset))
```

that seems to lead me to believe there could be even more clever way to handle it.

Reply:
Personally, not a fan of ternary operator. However, in this case, it does not hamper readability. Will modify the conditional